jetty-servlet/src/test/java/org/eclipse/jetty/servlet/ErrorPageTest.java

```diff
@@ -246,4 +308,59 @@ protected void doGet(HttpServletRequest request, HttpServletResponse response) t
             writer.println("getParameterMap()= " + request.getParameterMap());
         }
     }
+
+    public static class SingleDispatchFilter implements Filter
+    {
+        ConcurrentMap<Integer, Thread> dispatches = new ConcurrentHashMap<>();
+
+        @Override
+        public void init(FilterConfig filterConfig) throws ServletException
+        {
+
+        }
+
+        @Override
+        public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException
+        {
+            final Integer key = request.hashCode();
+            Thread current = Thread.currentThread();
+            final Thread existing = dispatches.putIfAbsent(key, current);
+            if (existing != null && existing != current)
+            {
+                System.err.println("DOUBLE DISPATCH OF REQUEST!!!!!!!!!!!!!!!!!!");
+                System.err.println("Thread " + existing + " :");
+                for (StackTraceElement element : existing.getStackTrace())
+                {
+                    System.err.println("\tat " + element);
+                }
+                IllegalStateException ex = new IllegalStateException();
+                System.err.println("Thread " + current + " :");
+                for (StackTraceElement element : ex.getStackTrace())
```

Comment:
Don't do this for exceptions, just `ex.printStackTrace()`.

Reply:
I wanted the exact same format as the dump above.... but now that it is not happening, I'll simplify.